tests/e2e/kafka_cluster_test.go

```diff
@@ -0,0 +1,57 @@
+// Copyright © 2023 Cisco Systems, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package e2e
+
+import (
+	"fmt"
+
+	"github.com/banzaicloud/koperator/api/v1beta1"
+	"github.com/gruntwork-io/terratest/modules/k8s"
+
+	. "github.com/onsi/ginkgo/v2"
+)
+
+const (
+	kafkaKind        = "kafkaclusters.kafka.banzaicloud.io"
+	kafkaClusterName = "kafka"
+)
+
+// createKafkaClusterIfDoesNotExist creates a kafka cluster if
+// there isn't a preexisting one
+func createKafkaClusterIfDoesNotExist(kubectlOptions *k8s.KubectlOptions, path string) {
+	By("Checking existing kafkaClusters")
+	err := checkExistenceOfK8sResource(kubectlOptions, kafkaKind, kafkaClusterName)
+
+	if err == nil {
+		fmt.Printf("Kafka cluster %s already exists", kafkaClusterName)
+	} else {
+		By("Deploying a KafkaCluster")
+		createK8sResourcesFromManifest(kubectlOptions, path, false)
+	}
+
+	return
+}
+
+// requireCreatingKafkaCluster creates a Kafka cluster and
+// checks the success of that operation.
+func requireCreatingKafkaCluster(kubectlOptions *k8s.KubectlOptions, path string) {
+	It("Creating a Kafka cluster", func() {
+		createKafkaClusterIfDoesNotExist(kubectlOptions, path)
+		By("Verifying Kafka cluster and it's pods")
+		waitK8sResourceCondition(kubectlOptions, kafkaKind, fmt.Sprintf("jsonpath={.status.state}=%s", string(v1beta1.KafkaClusterRunning)), "300s", "", kafkaClusterName)
+		// TODO: Can't really get this to work right now, might have to create a new function or update this one a bit
+		//requireRunningPods(kubectlOptions, "kafka_cr", "kafka-0", "kafka-1", "kafka-2")
```

Comment:
Here we should do a wait test: 
kubectl --namespace kafka wait pods --for=condition=Ready --timeout=5s --selector=kafka_cr=almafa